readme/api/tutorials/cm6_plugin.md

```diff
@@ -0,0 +1,622 @@
+```typescript
+// ...
+
+// Add this:
+const registerMessageListener = async (contentScriptId: string) => {
+	await joplin.contentScripts.onMessage(
+		contentScriptId,
+		
+		// Sending messages with `context.postMessage`
+		// from the content script with `contentScriptId`
+		// calls this onMessage listener:
+		async (message: any) => {
+			if (message === 'getSettings') {
+				const settingValue = await joplin.settings.value(highlightLineSettingId);
+				return {
+					highlightActiveLine: settingValue,
+				};
+			}
+		},
+	);
+};
+
+joplin.plugins.register({
+	onStart: async function() {
+		await registerSettings();
+
+		// Add this:
+		const contentScriptId = 'some-content-script-id';
+		await registerMessageListener(contentScriptId);
+		
+		await joplin.contentScripts.register(
+			ContentScriptType.CodeMirrorPlugin,
+			contentScriptId,
+			'./contentScript.js',
+		);
+	}
+});
+```
+
+### Get the setting from the content script
+
+Open `contentScript.ts` and update it with the following:
+```typescript
+import { lineNumbers, highlightActiveLine } from '@codemirror/view';
+
+// We're now using `context`: Rename it from `_context`
+// to `context`.
+export default (context: { contentScriptId: string, postMessage: any }) => {
+	return {
+		// An `async` was also added so that we can `await` the result of
+		// `context.postMessage`:
+		plugin: async (codeMirrorWrapper: any) => {
+			codeMirrorWrapper.addExtension(lineNumbers());
+
+			// Add this:
+			// Get settings from the main script with postMessage:
+			const settings = await context.postMessage('getSettings');
+			if (settings.highlightActiveLine) {
+				codeMirrorWrapper.addExtension(highlightActiveLine());
+			}
+		},
+	};
+};
+```
+
+Above, we get settings from `index.ts` by sending a message with the text "getSettings". This fires the `onMessage` listener that was registered earlier. Its return value is stored in the `settings` variable.
+
+Note that [`highlightActiveLine`](https://codemirror.net/docs/ref/#view.highlightActiveLine) is another built-in CodeMirror extension. It adds the `cm-activeLine` class to all lines that have a cursor on them.
+
+
+<details><summary>Alternative approach to getting settings: Registering an editor command</summary>
+
+Above, we use `postMessage` and `onMessage` to access settings.
+
+An alternative way to do this would be to register an editor command with code similar to the following:
+```typescript
+// You may need to add @codemirror/state to package.json
+import { Compartment } from '@codemirror/state';
+
+// ...
+
+	plugin: async (codeMirrorWrapper: any) => {
+		// See https://codemirror.net/examples/config/#compartments
+		const highlightExtension = new Compartment();
+		codeMirrorWrapper.addExtension(highlightExtension.of([]));
+
+		codeMirrorWrapper.defineExtension('myExtension__setHighlightActiveLine', (highlighted: boolean) => {
```

Comment:
Two notes:
- This may be broken in some cases until https://github.com/laurent22/joplin/pull/9928 is merged.
    - It fixes an issue with `defineExtension`.
- This is a CodeMirror 5 compatibility layer method. Currently, there is no CodeMirror 6 way to register an editor command.
- An alternative CodeMirror 5 way to register commands is
```ts
CodeMirror.commands.commandName = () => {
  // Command here
}
```

It might make more sense to use this for CodeMirror 6 or, alternatively, to add a new method for registering CodeMirror commands.

Reply:
https://github.com/laurent22/joplin/pull/9956 adds the `.registerCommand` API, so this should now be addressed!